ForumProject/subscriptions/tests.py

```diff
@@ -1,3 +1,386 @@
+            last_name='User2',
+            phone_number='+3801234562',
+            password='password2',
+            is_active=True
+        )
+
+        refresh_investor = RefreshToken.for_user(cls.investor_user2)
+        cls.investor_token2 = str(refresh_investor.access_token)
+
+        cls.startup_user = CustomUser.objects.create_user(
+            email='startup@example.com',
+            first_name='Startup',
+            last_name='User',
+            phone_number='+3809876543',
+            password='password',
+            is_active=True
+        )
+
+        refresh_startup = RefreshToken.for_user(cls.startup_user)
+        cls.startup_token = str(refresh_startup.access_token)
+
+    def setUp(self):
+        '''
+        Setup that is executed before each test case.
+        '''
+
+        self.client = APIClient()
+        # Set up credentials for the investor user
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.investor_token)
+        
+        # Create investor profile
+        self.investor_profile = Investor.objects.create(
+            investor_name='Test Investor',
+            investor_industry='IT',
+            investor_phone='+3801234567',
+            investor_country='UA',
+            investor_city='Test City',
+            investor_address='Test Address'
+        )
+        
+        # Associate investor profile with investor user and assign role
+        UserInvestor.objects.create(customuser=self.investor_user, investor=self.investor_profile, investor_role_id=1)
+        UserRoleCompany.objects.create(user=self.investor_user, role='investor', company_id=self.investor_profile.id)
+        
+        # Set up credentials for the investor user2
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.investor_token2)
+        
+        # Create investor profile
+        self.investor_profile2 = Investor.objects.create(
+            investor_name='Test Investor2',
+            investor_industry='IT',
+            investor_phone='+3801234567',
+            investor_country='UA',
+            investor_city='Test City2',
+            investor_address='Test Address2'
+        )
+        
+        # Associate investor profile with investor user and assign role
+        UserInvestor.objects.create(customuser=self.investor_user2, investor=self.investor_profile2, investor_role_id=1)
+        UserRoleCompany.objects.create(user=self.investor_user2, role='investor', company_id=self.investor_profile2.id)
+        
+        # Switch credentials to the startup user
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.startup_token)
+
+        # Create startup profile
+        self.startup_profile = Startup.objects.create(
+            startup_name='Test Startup',
+            startup_industry='Tech',
+            startup_phone='+3809876543',
+            startup_country='UA',
+            startup_city='Test City',
+            startup_address='Test Address'
+        )
```

Comment:
You are setting credentials multiple times for different users in `setUp()`, which might lead to unintended behavior or confusion about which user is authenticated in subsequent test methods. Make sure to set or reset credentials at the beginning of each test method if tests require specific user authentication

Reply:
added tearDown method,
unintended behavior or confusion is unlikely because the creation of each profile is accompanied by different tokens

Done, thanks!